src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2053,6 +2210,280 @@ mesh::generate_index_for_single_domain(const Node &mesh,
+{
+    index_t res = 0;
+
+    NodeConstIterator csitr = mesh["coordsets"].children();
+    while(csitr.has_next())
+    {
+        const Node &cset = csitr.next();
+        index_t cset_dim = cset["values"].number_of_children();
+        if (cset_dim > res)
+        {
+            res = cset_dim;
+        }
+    }
+
+    return res;
+}
+
+//-----------------------------------------------------------------------------
+bool
+mesh::is_dimension_consistent(const conduit::Node &mesh)
+{
+    bool started = false;
+    index_t dim = 0;
+
+    NodeConstIterator csitr = mesh["coordsets"].children();
+    while(csitr.has_next())
+    {
+        const Node &cset = csitr.next();
+        index_t cset_dim = cset["values"].number_of_children();
+        if (started && dim != cset_dim)
+        {
+            return false;
+        }
+        dim = cset_dim;
+        started = true;
+    }
+
+    return true;
+}
+
+//-----------------------------------------------------------------------------
+bool
+mesh::is_convertible_to_strip(const Node &mesh,
+                              Node &info)
+{
+    const std::string protocol = "convertible_to_strip";
+    bool res = true;
+    info.reset();
+
+    // Each coordset c in mesh["coordsets"] must have one child of c["values"]
+    bool consistent_dim = is_dimension_consistent(mesh);
+    if (!consistent_dim)
+    {
+        log::error(info, protocol, "inconsistent dimensionality in the mesh coordsets");
+        res = false;
+    }
+
+    index_t mesh_dim = max_dimension(mesh);
+    if (mesh_dim > 1)
+    {
+        log::error(info, protocol, "a mesh coordset has more than one child");
+        res = false;
+    }
+
+    // Only element-associated fields allowed (no vertex-fields)
+    // TODO Relax this requirement?
+    NodeConstIterator fitr = mesh["fields"].children();
+    while(fitr.has_next())
+    {
+        const Node &f = fitr.next();
+        // This method requires fields to have "association" == "element".
+        if (!f.has_child("association") || f["association"].as_string() != "element")
+        {
+            log::error(info,
+                       protocol,
+                       "fields[" + log::quote(fitr.name()) + "/association] != element");
+            res = false;
+        }
+    }
+
+    return res;
+}
+
+//-----------------------------------------------------------------------------
+bool
+mesh::is_convertible_to_one_dimension(const Node &mesh,
+                                      Node &info)
+{
+    const std::string protocol = "convertible_to_one_dimension";
+    bool res = true;
+    info.reset();
+
+    // Each coordset c in mesh["coordsets"] must have two children of c["values"]
+    bool consistent_dim = is_dimension_consistent(mesh);
+    if (!consistent_dim)
+    {
+        log::error(info, protocol, "inconsistent dimensionality in the mesh coordsets");
+        res = false;
+    }
+
+    index_t mesh_dim = max_dimension(mesh);
```

Comment:
We can use: `conduit::blueprint::mesh::utils::coordset::dims` for the specific coordset